web/src/api/ApiClient.ts

```diff
@@ -51,6 +51,33 @@ export class ApiClient {
             throw err
         }
     }
+
+    async register(
+        email: string,
+        password: string,
+        age: number,
+        gender: string
+    ) {
+        try {
+            return await this.baseRequest<void>(`user`, {
+                method: 'POST',
+                body: JSON.stringify({
+                    email,
+                    password,
+                    age,
+                    gender,
+                }),
+            })
+        } catch (err) {
+            if (
+                err instanceof FailedRequestError &&
+                err.response.status === 400
+            ) {
+                throw new EmailUsedError()
+            }
+            throw err
```

Comment:
It can also be "operation failed" error, for whatever reason... Backend informs which error it is in the response.